### PR TITLE
feat(neon): Add `JoinHandle` as a result of `Channel::send`

### PR DESCRIPTION
**Pull request by kjvalencik**
Part of https://github.com/neon-bindings/rfcs/pull/32

### CODE REVIEW

File: test/napi/lib/threads.js [hunk 1]
```diff
@@ -71,4 +71,17 @@ const assert = require('chai').assert;
     // Asynchronously GC to give the task queue a chance to execute
     setTimeout(() => global.gc(), 10);
   });
+
+  it('should be able to join on the result of a channel', function (cb) {
+    let msg = "Uninitialized";
```
Comment (dherman) [inline test/napi/lib/threads.js:76]: This and the Rust logic that goes with it took me a few minutes to follow 😂 

So `msg` is a shared mutable cell, we're modifying it after a 10ms timeout, and we're creating a Rust thread that waits 100ms and sends one event back to the main thread with the updated message, joins on that message, and sends another event back to the main thread that sends the message to an async callback. And then that callback tests that the message was the updated message.

Did I get that right? A comment might help 😅

**Review by dherman — APPROVED** [review]
Looks good to me! Maybe some comments in the test, which has kind of intricate control flow, would help. But it's up to you.

File: src/event/event_queue.rs [hunk 2]
```diff
@@ -115,21 +116,29 @@ impl Channel {
...
     ///
     /// See [`SendError`] for additional details on failure causes.
-    pub fn try_send<F>(&self, f: F) -> Result<(), SendError>
+    pub fn try_send<T, F>(&self, f: F) -> Result<JoinHandle<T>, SendError>
     where
-        F: FnOnce(TaskContext) -> NeonResult<()> + Send + 'static,
+        T: Send + 'static,
+        F: FnOnce(TaskContext) -> NeonResult<T> + Send + 'static,
     {
+        let (tx, rx) = mpsc::sync_channel(1);
```
[the remark this answers is not in the document]
Comment (kjvalencik) [reply src/event/event_queue.rs:124]: Yes, that's common naming. It comes from telecomm.

https://doc.rust-lang.org/rust-by-example/std_misc/channels.html

File: test/napi/lib/threads.js (hunk 1 above)
Comment (kjvalencik) [reply test/napi/lib/threads.js:76]: Yes, that's right! I'll write a more detailed comment that also explains _what_ we're trying to prove here. Specifically, we want to confirm that we can join on the result of `Channel::send` without impacting the JavaScript main thread.

Comment (kjvalencik) [reply test/napi/lib/threads.js:79]: Added a bunch of comments. 😄

**Pull request merged**